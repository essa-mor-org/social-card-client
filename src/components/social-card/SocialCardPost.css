.Social_post_text{
	color: #1d2129;
	font-size: 14px;
}

.Social_post_image{
	max-width: 500px;
	border-top: 1px solid rgb(223, 226, 229);
	border-bottom: 1px solid rgb(223, 226, 229);
	margin: 10px 0 10px 0;
}