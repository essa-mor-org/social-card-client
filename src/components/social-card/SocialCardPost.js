import React from 'react';
import PropTypes from 'prop-types';
import empty from 'is-empty';

import './SocialCardPost.css';

function SocialCardPost({ text, image }) {
	return (<div>
		{!empty(text) && <div className="Social_post_text">
			{text}
		</div>}
		<a href="#">
			<img className="Social_post_image" src={image} />
		</a>
	</div>);
}

SocialCardPost.propTypes = {
	text: PropTypes.string,
	image: PropTypes.string.isRequired
};

export default SocialCardPost;