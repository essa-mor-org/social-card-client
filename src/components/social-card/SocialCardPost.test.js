import React from 'react';
import SocialCardPost from './SocialCardPost';
import renderer from 'react-test-renderer';

describe('SocialCardPost test', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<SocialCardPost
				image="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/n.jpg"
				text="some text some text" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders correctly - no text', () => {
		const tree = renderer
			.create(<SocialCardPost
				image="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/n.jpg" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
