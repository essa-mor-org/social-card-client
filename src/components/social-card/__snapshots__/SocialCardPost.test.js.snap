// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialCardPost test renders correctly - no text 1`] = `
<div>
  <a
    href="#"
  >
    <img
      className="Social_post_image"
      src="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/n.jpg"
    />
  </a>
</div>
`;

exports[`SocialCardPost test renders correctly 1`] = `
<div>
  <div
    className="Social_post_text"
  >
    some text some text
  </div>
  <a
    href="#"
  >
    <img
      className="Social_post_image"
      src="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/n.jpg"
    />
  </a>
</div>
`;
